outbox-kafka-spring/src/main/java/one/tomorrow/transactionaloutbox/repository/DefaultOutboxEntityManager.java

```diff
@@ -15,25 +15,19 @@
  */
 package one.tomorrow.transactionaloutbox.repository;
 
-import lombok.AllArgsConstructor;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
 import org.springframework.stereotype.Component;
 
 
 @Component
-@AllArgsConstructor
-public class LegacyOutboxSessionFactory implements OutboxSessionFactory {
+public class DefaultOutboxEntityManager implements OutboxEntityManager {
```

Comment:
Do we still need the `OutboxEntityManager` and `DefaultOutboxEntityManager`, or could they be removed now?

Reply:
I'd be fine with removing them, using the `EntityManager` directly instead. I left it inside because I don't know if others who are using the transactional outbox maybe use different implementations of the `OutboxSessionFactory` ...

Reply:
@hnrkdmsk @danielrehmann Are there services in your context that are using the `LegacyOutboxSessionFactory`, without JPA set up? If so, how should they transition to the new implementation / are there plans for this?

Reply:
Btw, to have more context, this was the commit adding support for both hibernate _and_ JPA: https://github.com/tomorrow-one/transactional-outbox/commit/510ede5817b7c322995fbfad1f2f2bac4c75f890

Reply:
@magro I am unaware of services that use the Legacy approach besides the banking team. As the banking team is driving this change and explicitly want to remove this I would be fine with dropping the Legacy support here.

Reply:
I have the same opinion as @danielrehmann 👍🏼, but we should first mark it as deprecated or go with a breaking change. WDYT? 

Based on [this](https://stackoverflow.com/questions/5640778/hibernate-sessionfactory-vs-jpa-entitymanagerfactory), we don't have to offer a Hibernate specific solution if we don't use it internally any more.

Reply:
I've added another commit to drop the legacy support. The EntityManager is now used directly without any interface in between.

Reply:
Perfect! If there will be any requirement for an abstraction in the future we could still add it.